### PR TITLE
test(ext/node): Port _fs_writeFile_test.ts from deno_std

### PR DESCRIPTION
**Pull request by fbaltor**
Porting the `node/_fs/_fs_writeFile_test.ts` from [deno_std](https://github.com/denoland/deno_std/blob/0.177.0/node/_fs/_fs_writeFile_test.ts), one of the tasks of the bigger issue #17840.

### CODE REVIEW

**Comment by fbaltor** [comment]
There are two main points I'm having difficult with, which are:

1. **The `TextEncodings` type**, which is currently hardcoded in the test. The [original test](https://github.com/denoland/deno_std/blob/b4239898d6c6b4cdbfd659a4ea1838cf4e656336/node/_fs/_fs_writeFile_test.ts#L10) imports it from an utils file, but the correspondent one in the `deno` repo seems to be unreachable (`ext/node/polyfills/_utils.ts`, [link](https://github.com/fbaltor/deno/blob/72fe9bb47005e720444e65a66e91559287137780/ext/node/polyfills/_utils.ts#L9)).

2. **The test of cancellation with an `AbortSignal`** ([link](https://github.com/fbaltor/deno/blob/218ad587de27dcdc17635e39831fee92861d6259/cli/tests/unit_node/_fs/_fs_writeFile_test.ts#L237-L259)) that is not passing due to the following error:

`error: TS2719 [ERROR]: Type 'AbortSignal' is not assignable to type 'AbortSignal'. Two different types with this name exist, but they are unrelated.
`
image

File: cli/tests/unit_node/_fs/_fs_writeFile_test.ts [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+      ["utf8", "hello world"],
+      ["utf-8", "hello world"],
+    ];
+
+    for (const [encoding, value] of encodings) {
+      const res = await new Promise((resolve) => {
+        writeFile(
+          "_fs_writeFile_test_file.txt",
+          value,
+          encoding as TextEncodings,
```
Comment (bartlomieju) [inline cli/tests/unit_node/_fs/_fs_writeFile_test.ts:161]: What happens if you just skip `as TextEncodings` bit?

**Comment by bartlomieju** [comment]
@fbaltor as for no 2, you will have to use `as any` there because of the problem described in https://github.com/denoland/deno/issues/19527

File: cli/tests/unit_node/_fs/_fs_writeFile_test.ts (hunk 1 above)
Comment (fbaltor) [reply cli/tests/unit_node/_fs/_fs_writeFile_test.ts:161]: It complais: `Argument of type 'string' is not assignable to parameter of type 'WriteFileOptions'`. This parameter should be `Encodings | CallbackWithError | WriteFileOptions | undefined`, with type `Encodings = BinaryEncodings | TextEncodings`.

**Comment by kt3k** [comment]
@fbaltor Please run `tools/format.js` to pass CI checks

**Review by kt3k — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**